src/components/StatePicker.js

```diff
@@ -15,30 +16,57 @@ const propTypes = {
     label: PropTypes.string,
 
     /** A callback method that is called when the value changes and it received the selected value as an argument */
-    onChange: PropTypes.func.isRequired,
+    onInputChange: PropTypes.func.isRequired,
 
     /** The value that needs to be selected */
     value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
 
+    /** Indicates that the input is being used with the Form component */
+    isFormInput: PropTypes.bool,
+
+    /**
+    * The ID used to uniquely identify the input
+    *
+    * @param {Object} props - props passed to the input
+    * @returns {Object} - returns an Error object if isFormInput is supplied but inputID is falsey or not a string
+    */
+    inputID: props => FormUtils.validateInputIDProps(props),
+    
+    /** Saves a draft of the input value when used in a form */
+    shouldSaveDraft: PropTypes.bool,
+
+    /** Callback that is called when the text input is blurred */
+    onBlur: PropTypes.func,
+
+    /** Error text to display */
+    errorText: PropTypes.string,
+   
     ...withLocalizePropTypes,
 };
 
 const defaultProps = {
     label: '',
     value: '',
+    errorText: '',
+    shouldSaveDraft: false,
 };
 
-const StatePicker = props => (
+const StatePicker = forwardRef((props, ref) => (
     <Picker
+        ref={ref}
         placeholder={{value: '', label: '-'}}
         items={STATES}
-        onInputChange={props.onChange}
+        onInputChange={props.onInputChange}
         value={props.value}
         label={props.label || props.translate('common.state')}
         hasError={props.hasError}
```

Comment:
This should be removed. Right?

Reply:
I think we have to update the name only from onChange to onInputChange, if i am removing this then picker is not working

Reply:
I meant the hasError. Review comments refer to the line on which they are tagged.

Reply:
I have removed hasError from props only, let me remove hasError from here too